tools/clang/lib/Sema/SemaHLSL.cpp

```diff
@@ -15071,7 +15083,8 @@ bool Sema::DiagnoseHLSLDecl(Declarator &D, DeclContext *DC, Expr *BitWidth,
   }
 
   // Disallow long vecs from $Global cbuffers.
-  if (isGlobal && !isStatic && !isGroupShared) {
+  if (isGlobal && !isStatic && !isGroupShared &&
+      !IS_BASIC_RAWBUFFER(basicKind)) {
```

Comment:
The fact that you need this makes me think something else isn't detecting correctly.  We should not be looking at any objects that don't go into the constant buffer at all.  I think the property look up to use is `IS_BASIC_OBJECT`.

Note: when looking at the property table, I noticed some erroneous entries (not from your change).  I filed: #7293 for that follow-up.
```suggestion
      !IS_BASIC_OBJECT(basicKind)) {
```

Reply:
Sure enough! This simplifies things.

Reply:
I did see a lot of incorrect bits that I refrained from addressing for the time being. Many of them are never checked.